ast.c

```diff
@@ -50,6 +50,7 @@
 #define ast_register_flag_constant(name, value) \
 	REGISTER_NS_LONG_CONSTANT("ast\\flags", name, value, CONST_CS | CONST_PERSISTENT)
 
+// The number of cache slots there are must be the same as AST_NUM_CACHE_SLOTS in php_ast.h
```

Comment:
Just as a side node, we can drop these cache slots in favor of directly writing to OBJ_PROP, which is the pattern I prefer nowadays.

Reply:
Ok. I assume that works even with user-defined subclasses and traits? I'm only moderately familiar with how inheritance works in php for properties of classes

Reply:
That's more of a question about using that pattern in other PECLs or with obscure PECLs for debugging, though.

php-ast never returns subclasses instead of ast\Node in the latest releases.

Reply:
This is implemented and works, I'll target the release after 1.0.15